src/meta/meta_backup_service.cpp

```diff
@@ -1617,5 +1623,31 @@ void backup_service::start_backup_app(start_backup_app_rpc rpc)
     response.err = err;
 }
 
+void backup_service::query_backup_status(query_backup_status_rpc rpc)
+{
+    const query_backup_status_request &request = rpc.request();
+    query_backup_status_response &response = rpc.response();
+
+    int32_t app_id = request.app_id;
+    {
+        zauto_lock l(_lock);
+        for (const auto &backup : _backup_states) {
+            if (app_id == backup->get_backup_app_id()) {
+                response.backup_items.emplace_back(backup->get_backup_item());
+            }
+        }
+    }
+
+    if (response.backup_items.empty()) {
+        response.err = ERR_INVALID_PARAMETERS;
+        response.hint_message = fmt::format("No available backup for app {}.", app_id);
+        return;
+    }
+    response.__isset.backup_items = true;
+    response.hint_message = fmt::format(
+        "There are {} available backups for app {}.", response.backup_items.size(), app_id);
```

Comment:
What is the meaning of 'available backups'? Is `response.backup_items.size()` always equal to app partition_count?

Reply:
`response.backup_items.size()` is the number of backups of the app stored in meta memory.

Reply:
Well, I misunderstand before, and I raise another question. Will response contain many history backups for this app if meta server runs long time and this app have many backups? The rpc caller such as backup manager should analyse those backups to find out who is the real backup it wants.

Reply:
Yes. I added a optional field `backup_id` to help users get specific backup info. But if `backup_id` is not specified, meta will response with all backups for the requested app.